File: app/src/main/java/com/example/point/MainActivity.java
package com.example.point;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * @author YangJ
 */
public class MainActivity extends AppCompatActivity {

    // TAG
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initData();
        initView();
    }

    private void initData() {

    }

    private void initView() {
        Button button = findViewById(R.id.btn_click);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, ShowActivity.class));
                Log.i(TAG, "initView: " + v.getId());
            }
        });
        findViewById(R.id.btn_long_click).setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View v) {
                finish();
                Log.i(TAG, "onLongClick: " + v.getId());
                return true;
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.menu_add:
                Toast.makeText(this, R.string.menu_add, Toast.LENGTH_SHORT).show();
                break;
            case R.id.menu_del:
                Toast.makeText(this, R.string.menu_del, Toast.LENGTH_SHORT).show();
                break;
            default:
                break;
        }
        return true;
    }
}

File: app/src/main/java/com/example/point/MainActivity.kt
package com.example.point

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

/**
 * @author YangJ
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initData()
        initView()
    }

    private fun initData() {

    }

    private fun initView() {
        btn_click.setOnClickListener {
            startActivity(Intent(this, ShowActivity::class.java))
        }
        btn_long_click.setOnLongClickListener {
            finish()
            true
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item?.itemId) {
            R.id.menu_add -> {
                Toast.makeText(this, R.string.menu_add, Toast.LENGTH_SHORT).show()
            }
            R.id.menu_del -> {
                Toast.makeText(this, R.string.menu_del, Toast.LENGTH_SHORT).show()
            }
        }
        return true
    }
}
